File: language-support/js/daml-ledger/src/model/create_command.ts
// Copyright (c) 2019 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
// SPDX-License-Identifier: Apache-2.0

import * as ledger from '..';

/**
 * @interface CreateCommand
 * @memberof ledger
 */
export interface CreateCommand {
    /**
     * @member {ledger.Identifier} templateId
     * @memberof ledger.CreateCommand
     * @instance
     */
    templateId: ledger.Identifier
    /**
     * @member {ledger.Record} arguments
     * @memberof ledger.CreateCommand
     * @instance
     */
    arguments: ledger.Record
}

File: ledger-service/http-json-perf/src/main/scala/com/daml/http/perf/scenario/CreateCommand.scala
// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
// SPDX-License-Identifier: Apache-2.0
package com.daml.http.perf.scenario

import io.gatling.core.Predef._
import io.gatling.http.Predef._

@SuppressWarnings(Array("org.wartremover.warts.NonUnitStatements"))
class CreateCommand extends Simulation with SimulationConfig {

  private val jsonCommand = """{
  "templateId": "Iou:Iou",
  "payload": {
    "issuer": "Alice",
    "owner": "Alice",
    "currency": "USD",
    "amount": "9.99",
    "observers": []
  }
}"""

  private val request = http("CreateCommand")
    .post("/v1/create")
    .body(StringBody(jsonCommand))

  private val scn = scenario("CreateCommandScenario")
    .repeat(1000)(exec(request))

  setUp(
    scn.inject(atOnceUsers(1))
  ).protocols(httpProtocol)
}
